actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -75,10 +75,12 @@ def formats
 
       # Sets the \variant for template.
       def variant=(variant)
-        if variant.is_a?(Symbol)
-          @variant = [variant]
-        elsif variant.nil? || variant.is_a?(Array) && variant.any? && variant.all?{ |v| v.is_a?(Symbol) }
+        if variant.nil?
           @variant = variant
```

Comment:
I might be mistaken here, but I think these if statements can be simplified as:

```ruby
variants = Array(variants).compact

if variants.all? { |v| v.is_a?(Symbol) }
  @variant = VariantInquirer.new(variants)
else
  raise ArgumentError, "super long message"
end
```

Though I don't know how much of a hot spot this code path is.